gencode.c

```diff
@@ -1552,7 +1585,7 @@ gen_load_macplrel(offset, size)
 		 * as an offset.
 		 */
 		s2 = new_stmt(BPF_LD|BPF_IND|size);
-		s2->s.k = offset;
+		s2->s.k = off_macpl + offset;
```

Comment:
Perhaps I should have pulled this out into a separate commit but this is actually fixing a bug with all variable length headers and should not change the intended meaning of what is there currently.

The problem is that some filters (like VLAN) increment off_macpl as they are parsed. In the fixed MAC header case, this is handled properly since it is just incorporated into the header length. However, currently with variable length headers we only look at the variable component and ignore the fixed part, so the VLAN stack can't be handled. This is changing it so that we look at both pieces for variable headers; in cases where there are no VLAN filters, off_macpl is zero so things still work.

I don't know of any filter that increments off_ll but it seemed right to apply the same logic there.